robsize.cc

```diff
@@ -416,56 +421,85 @@ void print_tests() {
     }
 }
 
-/** handle the arguments, return true if everything OK */
-bool handle_args(int argc, const char *argv[]) {
-    int firstopt = 2;
-    if (argc >= 2) {
-        if (sscanf(argv[1], "%d", &instr_type) == 0) {
-            firstopt = 1;
-        }
-    }
-    for (int i = firstopt; i < argc; i++)
-    {
-        if (!strcmp(argv[i], "--fast")) {
-            its >>=2;
-            outer_its >>=2;
-        } else if (!strcmp(argv[i], "--help")) {
-            print_usage();
-            exit(EXIT_SUCCESS);
-        } else if (!strcmp(argv[i], "--list")) {
-            print_tests();
-            exit(EXIT_SUCCESS);
-        } else if (!strcmp(argv[i], "--csv")) {
-            plot_mode = true;
-        } else if (!strcmp(argv[i], "--slow")) {
-            outer_its <<=1;
-        } else if (!strcmp(argv[i], "--superfast")) {
-            its >>=4;
-            outer_its >>=3;
-        } else if (!strcmp(argv[i], "--write-asm")) {
-            // print the generated instructions to a file and quit
-            print_ibuf = true;
-        } else {
-            fprintf(stderr, "Uncognized argument: %s\n", argv[i]);
-            return false;
+/* Command line options for getopt_long() */
+static struct option long_options[] = {
+    {"help",       no_argument, NULL, 'h'},
+    {"list",       no_argument, NULL, 'h'},
```

Comment:
`h` short arg clashes with --help, should be `l`?

Reply:
I did not realize `print_test()` and `print_usage()` were two different methods so I folded the cases into one. Sorry about that, correcting it.

Reply:
It would be OK to merge them into one too, but then just get rid of `--list` and put it all in `--help`.

Reply:
Should be separated out now. The output for list is quite large so it should be easier to parse usage and list of tests if they were separate.